controllers/scripts/create_pod_status_patch.py

```diff
@@ -374,132 +380,150 @@ def init_volumes(pod_name, config):
-            logging.debug(f"In Filesystem initialization: {volume}")
-            if not os.path.exists(volume.get_mount_point()):
-                logging.error(f"pod-name: {pod_name} volume-name: {volume.volume_name} - Mounting point "
-                              f"does not exists")
-                raise FileNotFoundError(f"{volume} Volume path not found")
+                elif volume.effective_init_method == "none":
+                    logging.info(f"{volume} - Pass through")
+                else:
+                    logging.error(f"{volume} - Has invalid effective method")
+                    raise ValueError(f"{volume} - Has invalid effective method")
+
+            elif volume.volume_mode == "Filesystem":
+                logging.debug(f"In Filesystem initialization: {volume}")
+                if not os.path.exists(volume.get_mount_point()):
+                    logging.error(f"pod-name: {pod_name} volume-name: {volume.volume_name} - Mounting point "
+                                  f"does not exists")
+                    raise FileNotFoundError(f"{volume} Volume path not found")
 
-            if volume.effective_init_method == "deleteFiles":
+                if volume.effective_init_method == "deleteFiles":
 
-                find = "find {volume_path} -type f -delete".format(
-                    volume_path=volume.get_mount_point())
-                execute(find)
-                logging.info(f"{volume} - Initialized")
+                    find = "find {volume_path} -type f -delete".format(
+                        volume_path=volume.get_mount_point())
+                    execute(find)
+                    logging.info(f"{volume} - Initialized")
 
-            elif volume.effective_init_method == "none":
-                logging.info(f"{volume} - Pass through")
+                elif volume.effective_init_method == "none":
+                    logging.info(f"{volume} - Pass through")
+                else:
+                    logging.error(f"{volume} - Has invalid effective method")
+                    raise ValueError(f"{volume} - Has invalid effective method")
             else:
-                logging.error(f"{volume} - Has invalid effective method")
-                raise ValueError(f"{volume} - Has invalid effective method")
-        else:
-            logging.error(f"{volume} - Invalid volume-mode: {volume.volume_mode}")
-            raise ValueError(f"pod-name: {pod_name} - Invalid volume-mode: {volume.volume_mode}")
+                logging.error(f"{volume} - Invalid volume-mode: {volume.volume_mode}")
+                raise ValueError(f"pod-name: {pod_name} - Invalid volume-mode: {volume.volume_mode}")
 
         logging.debug(f"{volume} - Added to initialized-volume list")
         volumes.append(volume.volume_name)
 
-    logging.debug(f"{volumes} - Extending initialized-volume list")
-    volumes.extend(initialized_volumes)
+        for future in concurrent.futures.as_completed(fs=futures):
```

Comment:
Can you please check if this is enough for error handling? For example, what will happen if one of the executors raises an error?

Reply:
whatever exceptions executor.Submit will raise, we are storing them in future array and here line 448 all errors are being handled.